src/scikit_build_core/settings/metadata.py

```diff
@@ -1,50 +1,21 @@
 from __future__ import annotations
 
-import warnings
-from pathlib import Path
+import importlib
```

Comment:
One benefit of dropping entry points is that we can remove the dependency on importlib-metadata for older versions of Python, and get rid of it from the .._compat/importlib.py as well. It is still a test time dependency for the dynamic `test_pep517_wheel()` function, but only in the virtualenv that the wheel is installed into.

Reply:
I wouldn't be surprised if it sneaks in eventually for something (adding cmake modules likely will be done through plugins), but yes, for now.